lib/plugins/aws/deploy/compile/events/apiGateway/lib/apiKeys.js

```diff
@@ -39,9 +39,9 @@ module.exports = {
         _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
           newApiKeyObject);
 
-        // Add API Key to Outputs section
+        // Add Id of API Key to Outputs section (Note: This is the Id, not the value!)
         const newOutput = {
```

Comment:
I think we can easily do this filtering by inspecting the `provider.apiKey` array so that we'd get the API keys names for this service and filter all the API Keys that we got from AWS API. 

This is more straightforward imo. I just fear that adding extra outputs could be making deployment slower and increasing template size. Not to mention that it's pretty redundant